docs/processes/fuzzer_targets.md

```diff
@@ -0,0 +1,119 @@
+
+### Guidelines and tips
+
+*   Leave all randomness to the fuzzer. If a random selection of any kind is needed (e.g., of a locale), then use bytes from the fuzzer data to make the selection ([example](https://github.com/unicode-org/icu/blob/master/icu4c/source/test/fuzzer/break_iterator_fuzzer.cpp)).
+*   In many cases ICU unit tests can provide seed data or at least ideas for seed data. If the API under test requires a Unicode string then make sure that the seed data is in UTF-16 encoding. This can be achieved with e.g. the 'iconv' command or using an editor that saves test in UTF-16 command.
+
+### How to locally reproduce fuzzer findings
+
+At this time reproduction of fuzzer findings requires Docker installed on the local machine and the OSS-Fuzz project downloaded in a locale git client.
+
+1.  Install Docker (Ubuntu):
+
+    ```
+    sudo apt-get install docker
+    ```
+2.  Download OSS-Fuzz, switch into directory oss-fuzz/
+
+    In a git client directory, download the fuzzer system.
+
+    ```
+    git clone https://github.com/google/oss-fuzz.git
+    cd oss-fuzz/
+    ```
+3.  Build the Docker image for ICU.
+    In some setups root permissions may be required to connect to the Docker.
+
+    ```
+    [sudo] python infra/helper.py build_image icu
+    ```
+4.  Build the ICU fuzzers:
+
+    ```
+    [sudo] python infra/helper.py build_fuzzers --sanitizer [address | memory | undefined] icu
+    ```
+    Check that the fuzzer targets were build successfully: ```ls -l build/out/icu```
+
+5.   Reproduce the fuzzer finding.
+     First, get the testdata the fuzzer used when finding the issue. In the fuzzer bug report look for 'Reproducer Testcase', a click on the link will download the testdata. Then execute
+
+     ```
+     [sudo] python infra/helper.py reproduce icu <icu_fuzzer> <testdata>
+     ```
+     Concrete example:
+
+     ```
+     sudo python infra/helper.py reproduce icu uregex_open_fuzzer  ~/Downloads/clusterfuzz-testcase-minimized-uregex_open_fuzzer-5732067058384896
+     ```
+
+**Limitations:** When reproducing a fuzzer finding in the way outlined above the fuzzer environment will use the current ICU trunk from https://github.com/unicode-org/icu.git. Thus it is not possible to modify the code to try out a possible fix. What can be done is to redirect Docker to download ICU from a forked ICU repository. Open the file oss-fuzz/projects/icu/Dockerfile and adjust the line with ```git clone --depth 1 https://github.com/unicode-org/icu.git icu``` accordingly. Then modify the code in the forked repository and follow the steps above beginning with step 3, create a Docker image.
+
+This of course is still a tedious way of reproducing and working on a fuzzer finding. Ticket [ICU-20734](https://unicode-org.atlassian.net/browse/ICU-20734) aims to introduce a fuzzer driver that can reproduce certain fuzzer findings in a local ICU
```

Comment:
unfinished sentence? end with "... in a local ICU workspace." ?

Reply:
Yes, workspace will do here.